core/src/test/scala/org/apache/spark/eventhubs/EventHubsConfSuite.scala

```diff
@@ -207,6 +208,46 @@ class EventHubsConfSuite extends FunSuite with BeforeAndAfterAll {
     assert(actual.equals(expected))
   }
 
+  test("receiverListener serialization / deserialization") {
```

Comment:
More test cases are needed. Current test cases cover serialization/deserialization for both the sender and receiver but do not test the correctness of the collected metrics. At least two more test cases (one for the sender, one for the receiver) should be added to test the correctness of the collected metrics.

Reply:
How to test correctness of collected metrics? we don't have a mocked eventHub server to test against, or I can only generate some test metrics, pass in the metric to the listener, and verify the value in listener are equal to the passed-in value?

Reply:
You can use the EventHubsTestUtils class to create a simulated, in-memory Event Hubs instance. You can send a fixed number/size of events to the simulated instance, then receive those events and make sure the collected metrics during both send/receive are as expected.

Reply:
OK, thanks!